test/ephemeral-integration-test.sh

```diff
@@ -0,0 +1,222 @@
+#!/usr/bin/env bash
+#
+# Integration test runner for Konflux ephemeral namespace
+#
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+echo "=== Gabi Integration Test on Konflux ephemeral namespace ==="
+echo ""
+
+echo "Image URL: ${IMAGE_URL}"
+echo "Image Digest: ${IMAGE_DIGEST}"
+
+# Step 1: Deploy supporting services (database and mock-splunk)
+echo ""
+echo "Step 1: Deploying database and mock-splunk..."
+
+# Set the mock-splunk image to the locally built image
+MOCK_SPLUNK_IMAGE="${IMAGE_URL}"
+echo "Using mock-splunk image: ${MOCK_SPLUNK_IMAGE}"
+sed "s|{{MOCK_SPLUNK_IMAGE}}|${MOCK_SPLUNK_IMAGE}|g" test/test-pod.yml | oc apply -f -
+
+# Step 2: Wait for supporting services to be ready
+echo ""
+echo "Step 2: Waiting for services to be ready..."
+echo "Waiting for test-pod to be ready..."
+echo ""
+
+# Show progress while waiting
+(
+  while oc get pod/test-pod -o json 2>/dev/null | jq -e '.status.conditions[] | select(.type=="Ready" and .status=="False")' > /dev/null 2>&1; do
+    # Show container status
+    echo -n "$(date '+%H:%M:%S') - Pod status: "
+    oc get pod/test-pod -o jsonpath='{.status.phase}' 2>/dev/null || echo "Unknown"
+
+    # Show readiness status for each container
+    echo -n "  Containers ready: "
+    oc get pod/test-pod -o jsonpath='{range .status.containerStatuses[*]}{.name}={.ready} {end}' 2>/dev/null || echo "checking..."
+    echo ""
+
+    sleep 10
+  done
+) &
+PROGRESS_PID=$!
+
+# Wait for pod to be ready (increased timeout for Splunk)
+if oc wait --for=condition=ready pod/test-pod --timeout=60s; then
+    kill $PROGRESS_PID 2>/dev/null || true
+    wait $PROGRESS_PID 2>/dev/null || true
+    echo ""
+    echo "✅ Services are ready!"
+    oc get pod/test-pod
+    oc get service/test-pod
+else
+    kill $PROGRESS_PID 2>/dev/null || true
+    wait $PROGRESS_PID 2>/dev/null || true
+    echo ""
+    echo "❌ Error: Pod failed to become ready within 60 seconds"
+    echo ""
+    echo "Pod description:"
+    oc describe pod/test-pod
+    echo ""
+    echo "Container logs:"
+    oc logs test-pod --all-containers=true || true
+    exit 1
+fi
+
+# Verify service endpoints are available
+echo ""
+echo "Verifying service endpoints..."
+oc get endpoints test-pod
+
+# Step 3: Create a temporary pod manifest for running tests
+echo ""
+echo "Step 3: Creating test runner pod..."
+cat > /tmp/gabi-integration-test-pod.yaml <<EOF
+apiVersion: v1
+kind: Pod
```

Comment:
A job is better suited for this test-runner, see https://github.com/app-sre/shared-pipelines/blob/main/pipeline-tests/test-job-template.yaml